sparkmagic/sparkmagic/livyclientlib/sendtosparkcommand.py

```diff
@@ -0,0 +1,50 @@
+from sparkmagic.livyclientlib.command import Command
+from sparkmagic.livyclientlib.exceptions import BadUserDataException
+import sparkmagic.utils.constants as constants
+
+from abc import abstractmethod
+
+class SendToSparkCommand(Command):
+    def __init__(self, input_variable_name, input_variable_value, output_variable_name, spark_events=None):
+        super(SendToSparkCommand, self).__init__("", spark_events)
+        self.input_variable_name = input_variable_name
+        self.input_variable_value = input_variable_value
+        self.output_variable_name = output_variable_name
+
+    def execute(self, session):
+        try:
+            command = self.to_command(session.kind, self.input_variable_name, self.input_variable_value, self.output_variable_name)
+            (success, result) = command.execute(session)
+            result = self.input_variable_value #todo ISSUE#412
+            if not success:
```

Comment:
I wonder what kind of issues would result in not success and whether it's up to each implementation of this abstract class to handle the failures (e.g. JSON serialization failures could only be detected by commands that JSON serialize) and etcetera. What do you think?

Reply:
Why check for success here instead of simply passing it up?

Reply:
This section is heavily work-in-progressish and I literally have no idea why I did that, lol. Will surely refactor this part later. :)